private/buf/bufformat/formatter.go

```diff
@@ -259,6 +259,11 @@ func (f *formatter) writeFileHeader() {
 		if i == 0 && f.previousNode != nil && !f.leadingCommentsContainBlankLine(importNode) {
 			f.P("")
 		}
+
+		if i > 0 && importNode.Name.AsString() == importNodes[i-1].Name.AsString() {
```

Comment:
Can you add a comment as to what this is doing, ie that this removes duplicates because the importNodes are already sorted?

Reply:
@bufdev just added testdata and comment